lib/puppet/datatypes/impl/error.rb

```diff
@@ -1,39 +1,37 @@
 class Puppet::DataTypes::Error
-  attr_reader :message, :kind, :issue_code, :partial_result, :details
+  attr_reader :msg, :kind, :issue_code, :details
+  alias message msg
 
   def self.from_asserted_hash(hash)
-    new(hash['message'], hash['kind'], hash['issue_code'], hash['partial_result'], hash['details'])
+    new(hash['msg'], hash['kind'], hash['details'], hash['issue_code'])
   end
 
   def _pcore_init_hash
-    result = { 'message' => @message }
+    result = { 'msg' => @msg }
     result['kind'] = @kind unless @kind.nil?
-    result['issue_code'] = @issue_code unless @issue_code.nil?
-    result['partial_result'] = @partial_result unless @partial_result.nil?
     result['details'] = @details unless @details.nil?
+    result['issue_code'] = @issue_code unless @issue_code.nil?
     result
   end
 
-  def initialize(message, kind = nil, issue_code = nil, partial_result = nil, details = nil)
-    @message = message
+  def initialize(msg, kind = nil, details = nil, issue_code = nil)
```

Comment:
There'd be fewer test changes if you preserved `issue_code` then `details`. Any reason you reordered it?

Reply:
Yes details are widely used and part of the task API. issue_code is not. Users will often want to provide details to plan errors but most of the time there is little reason to use issue_code in addition to kind.

Reply:
I think `Result` needs to be updated to use this call signature. We should also probably have a test that exercises `error` of the Error type.

Reply:
Ah, ignore this comment. That'll be handled in Bolt when we update.